drivers/sensor/aht20/aht20.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Rafael Lee <rafaellee.img@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT asair_aht20
+
+#include <kernel.h>
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include <sys/crc.h>
+
+#include "aht20.h"
+
+LOG_MODULE_REGISTER(AHT20, CONFIG_SENSOR_LOG_LEVEL);
+
+static int aht20_channel_get(const struct device *dev, enum sensor_channel chan,
+			     struct sensor_value *val)
+{
+	struct aht20_data *drv_data = dev->data;
+	float temperature;
+
+	switch (chan) {
+	case SENSOR_CHAN_HUMIDITY:
+		val->val1 = 0;
+		val->val2 = (drv_data->humidity) * 1000000LL / (1 << 20);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temperature = (float)drv_data->temperature / (1 << 20) * 200 - 50.0;
+		val->val1 = (int32_t)temperature;
+		val->val2 = (temperature - val->val1) * 1000000LL;
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static int aht20_init(const struct device *dev)
+{
+	struct aht20_data *drv_data = dev->data;
+
+	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+	if (drv_data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!", DT_INST_BUS_LABEL(0));
+		return -EINVAL;
+	}
+
+	k_busy_wait(40000); // wait for 40ms
```

Comment:
reason? this will also delay system startup

Reply:
The sensor needs 40ms to initialize before any operation. Is there any good way to this during or after startup?

Reply:
Replaced with k_sleep